src/eredis.erl

```diff
@@ -40,36 +51,45 @@ start_link(Host, Port) ->
 start_link(Host, Port, Database) ->
     start_link(Host, Port, Database, "").
 
-start_link(Host, Port, Database, Password) ->
-    start_link(Host, Port, Database, Password, 100).
+start_link(Host, Port, Database, MaybeCredentials) ->
+    Credentials = make_credentials(MaybeCredentials),
+    start_link(Host, Port, Database, Credentials, 100).
 
-start_link(Host, Port, Database, Password, ReconnectSleep) ->
-    start_link(Host, Port, Database, Password, ReconnectSleep, ?TIMEOUT).
+start_link(Host, Port, Database, MaybeCredentials, ReconnectSleep) ->
+    Credentials = make_credentials(MaybeCredentials),
+    start_link(Host, Port, Database, Credentials, ReconnectSleep, ?TIMEOUT).
 
-start_link(Host, Port, Database, Password, ReconnectSleep, ConnectTimeout) ->
-    start_link(Host, Port, Database, Password, ReconnectSleep, ConnectTimeout, []).
+start_link(Host, Port, Database, MaybeCredentials, ReconnectSleep, ConnectTimeout) ->
+    Credentials = make_credentials(MaybeCredentials),
+    start_link(Host, Port, Database, Credentials, ReconnectSleep, ConnectTimeout, []).
 
-start_link(Host, Port, Database, Password, ReconnectSleep, ConnectTimeout, Options)
+start_link(Host, Port, Database, MaybeCredentials, ReconnectSleep, ConnectTimeout, Options)
```

Comment:
Nit: since we're adding a new arity to this function, maybe it would be a good opportunity to instead make it receive a single map argument and refactor the other clauses?